depsync/depsync.go

```diff
@@ -71,33 +87,43 @@ func New(pub pubsub.PublishSubscriber, db *bolt.DB, opts ...Option) (Syncer, err
 		publisher: pub,
 		conf:      conf,
 		startSync: make(chan bool),
+		syncNow:   make(chan bool),
+
+		// TODO: hard coded right now for initial testing
+		assigners: []autoassigner{autoassigner{filter: "*", profile_uuid: "95124484B114B710"}},
 	}
 
+	// apply our supplied options
 	for _, opt := range opts {
 		opt(sync)
 	}
 
+	// if no logger option has been set use the null logger
+	if sync.logger == nil {
```

Comment:
Add it before `opt(sync)` is called and then you don't need this check.

Reply:
But then I waste the resources of assigning/initializing a resource unnecessarily, no?